client/my-sites/checkout/checkout-thank-you/hundred-year-plan-thank-you/index.tsx

```diff
@@ -0,0 +1,172 @@
+import page from 'page';
+import React, { useEffect } from 'react';
+import WordPressLogo from 'calypso/components/wordpress-logo';
+import { useSelector, useDispatch } from 'calypso/state';
+import { fetchReceipt } from 'calypso/state/receipts/actions';
+import { getReceiptById } from 'calypso/state/receipts/selectors';
+import isUserRegistrationDaysWithinRange from 'calypso/state/selectors/is-user-registration-days-within-range';
+import { hideMasterbar } from 'calypso/state/ui/actions';
+
+interface Props {
+	siteSlug: string;
+	receiptId: number;
+}
+
+const PageContainer = styled.div``;
+const MasterBar = styled.div`
+	height: 48px;
+	width: 100%;
+	padding: 24px 0 0 24px;
+`;
+
+const Header = styled.h1< { isMobile: boolean } >`
+	font-size: ${ ( { isMobile } ) => ( isMobile ? '2rem' : '2.75rem' ) };
+	line-height: ${ ( { isMobile } ) => ( isMobile ? '32px' : '52px' ) };
+	text-align: center;
+	margin: 16px 0;
+`;
+
+const Content = styled.div< { isMobile: boolean } >`
+	margin: 0 auto;
+	padding: 24px ${ ( { isMobile } ) => ( isMobile ? '9px' : '24px' ) };
+	color: var( --studio-gray-5 );
+	max-width: min( 95vw, 877px );
+	text-align: center;
+	.hundred-year-plan-thank-you__thank-you-text-container {
+		margin: 24px ${ ( { isMobile } ) => ( isMobile ? '24px' : '80px' ) };
+	}
+`;
+
+const Highlight = styled.div`
+	margin-bottom: 32px;
+	text-align: center;
+	font-size: 16px;
+	p {
+		margin: 0;
+	}
+`;
+
+const ButtonBar = styled.div< { isMobile: boolean } >`
+	margin-bottom: ${ ( { isMobile } ) => ( isMobile ? '8px' : '32px' ) };
+	display: flex;
+	justify-content: center;
+	gap: 16px;
+	flex-direction: ${ ( { isMobile } ) => ( isMobile ? 'column' : 'row' ) };
+`;
+
+const StyledButton = styled( Button )`
+	border-radius: 4px;
+	box-shadow: 0px 1px 2px 0px rgba( 0, 0, 0, 0.05 );
+	font-size: 14px;
+	font-weight: 500;
+	line-height: 20px;
+	letter-spacing: 0.32px;
+	text-align: center;
+	background-color: none;
+	background: none;
+	color: var( --studio-gray-0 );
+`;
+const StyledLightButton = styled( StyledButton )`
+	border: 1px solid var( --gray-gray-0, #f6f7f7 );
+	background-color: var( --studio-black );
+	background: linear-gradient( #c1c0d3, #e3e2f3, #c1c0d3 );
+	color: var( --studio-black );
+`;
+
+const CustomizedWordPressLogo = styled( WordPressLogo )`
+	margin: 0;
+	fill: var( --studio-white );
+`;
+
+export default function HundredYearPlanThankYou( { siteSlug, receiptId }: Props ) {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const isUserJoinedWithinADay = useSelector( ( state ) =>
+		isUserRegistrationDaysWithinRange( state, null, 0, 1 )
+	);
+	const receipt = useSelector( ( state ) => getReceiptById( state, receiptId ) );
+	const isReceiptLoading = ! receipt.hasLoadedFromServer || receipt.isRequesting;
+	useEffect( () => {
+		dispatch( hideMasterbar() );
+		if ( isReceiptLoading && receiptId ) {
+			dispatch( fetchReceipt( receiptId ) );
+		}
+	}, [ dispatch, isReceiptLoading, receiptId ] );
```

Comment:
Non blocking/Nit: `hideMasterbar` will be dispatched two times - when `isReceiptLoading` is `true` and then when `isReceiptLoading` is `false`. We could separate these into two `useEffect` hook calls.